internal/core/vulndb/github_exploit_service.go

```diff
@@ -0,0 +1,166 @@
+package vulndb
+
+import (
+	"archive/zip"
+	"context"
+	"encoding/json"
+	"io"
+	"log/slog"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/l3montree-dev/flawfix/internal/database"
+	"github.com/l3montree-dev/flawfix/internal/database/models"
+)
+
+type githubExploitDBService struct {
+	exploitRepository exploitRepository
+	httpClient        *http.Client
+}
+
+type githubExploitRepository interface {
+	SaveBatch(tx database.DB, exploits []models.Exploit) error
+}
+
+func NewGithubExploitDBService(exploitRepository githubExploitRepository) githubExploitDBService {
+	return githubExploitDBService{
+		exploitRepository: exploitRepository,
+		httpClient:        &http.Client{},
+	}
+}
+
+var githubExploitDBURL = "https://github.com/nomi-sec/PoC-in-GitHub/archive/refs/heads/master.zip"
+
+func (s *githubExploitDBService) fetchGithubExploits(ctx context.Context) error {
+	// download exploits from the github repository
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, githubExploitDBURL, nil)
+	if err != nil {
+		slog.Info("Error while downloading the file %s", err)
+		return err
+	}
+
+	res, err := s.httpClient.Do(req)
+	if err != nil {
+		slog.Info("Error while downloading the file %s", err)
+		return err
+	}
+
+	defer res.Body.Close()
+
+	// Create the file
+	zipFile, err := os.Create("exploits.zip")
+	if err != nil {
+		slog.Info("Error while creating the file %s", err)
+		return err
+	}
+	defer zipFile.Close()
+
+	// Write the body to file
+	_, err = io.Copy(zipFile, res.Body)
+	if err != nil {
+		slog.Info("Error while writing the file %s", err)
+		return err
+	}
+
+	// Unzip the file
+	zipListing, err := zip.OpenReader("exploits.zip")
+	if err != nil {
+		slog.Info("Error while unzipping the file %s", err)
+		return err
+	}
+
+	defer zipListing.Close()
```

Comment:
```suggestion
	b, err := io.ReadAll(res.Body)
	if err != nil {
		slog.Info("error while reading the response body", "err", err)
		return err
	}

	defer res.Body.Close()

	// Unzip the file
	zipListing, err := zip.NewReader(bytes.NewReader(b), int64(len(b)))
	if err != nil {
		slog.Info("Error while unzipping the file %s", "err", err)
		return err
	}
```

We need all of the bytes in memory - since we are trying to unzip, which is not possible using a stream of bytes. Thus we can just keep the bytes in memory and unzip them.

This way, we avoid a lot of disk hits for writing and reading the file.